Library/Homebrew/cmd/uninstall.rb

```diff
@@ -9,15 +10,41 @@ def uninstall
       ARGV.kegs.each do |keg|
         keg.lock do
           puts "Uninstalling #{keg}... (#{keg.abv})"
+
+          # Find old name and old name symlinks for formula and remove them
+          begin
+            formula = Formulary.from_rack(keg.rack)
```

Comment:
Do we need to load formula at all. Can we do something like:

```ruby
HOMEBREW_CELLAR.subdirs.select { |keg| keg.symlink? && keg.resloved_path.to_s == keg.to_s }
```

This way seems to be more robust.